tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/TelemetryTests.cs

```diff
@@ -79,8 +82,10 @@ public void WhenDisabled_DoesntSendTelemetry()
             {
                 Assert.True(processResult.ExitCode == 0, $"Process exited with code {processResult.ExitCode}");
 
-                var spans = agent.WaitForSpans(expectedSpanCount, operationName: expectedOperationName);
-                Assert.Equal(expectedSpanCount, spans.Count);
+                var spans = agent.WaitForSpans(2);
+
+                Assert.Contains(spans, span => span.Name == expectedOperationName);
+                Assert.Contains(spans, span => span.Name == spanFromActivityOperationName);
```

Comment:
I'd definitely prefer we use snapshots here, to make sure all the parents etc line up.

Reply:
well both spans doesn't share the same traceId.... They are not part of the same execution context, and we are not testing the context propagation. (Also, we don't have an API to set the context of an activity yet).

Reply:
>They are not part of the same execution context, and we are not testing the context propagation. (Also, we don't have an API to set the context of an activity yet), only W3C is supported automatically

Hmmm, I feel like we should have integration tests for that 🤔 Definitely happy for that to be a new PR though!